src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/jobs/LicenceActivationService.kt

```diff
@@ -0,0 +1,94 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.jobs
+
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.Licence
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.LicenceRepository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.LicenceService
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonerSearchApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonerSearchPrisoner
+import java.time.LocalDate
+
+@Service
+class LicenceActivationService(
+  private val licenceRepository: LicenceRepository,
+  private val licenceService: LicenceService,
+  private val prisonApiClient: PrisonApiClient,
+  private val prisonerSearchApiClient: PrisonerSearchApiClient
+) {
+
+  @Transactional
+  fun licenceActivationJob() {
+    val potentialLicences = licenceRepository.getApprovedLicencesOnOrPassedReleaseDate()
+    val (eligibleLicences, ineligibleLicences) = determineLicenceEligibility(potentialLicences)
+    val licencesToActivate = determineEligibleLicencesToActivate(eligibleLicences)
+
+    if (licencesToActivate.any()) {
```

Comment:
No point checking here, the lower level will safely handle what to do if there are no licences.